crd/src/lib.rs

```diff
@@ -63,29 +80,84 @@ impl Display for KafkaVersion {
     }
 }
 
+#[derive(EnumIter, Debug, Display, PartialEq, Eq, Hash)]
+pub enum KafkaRole {
+    Broker,
+}
+
 #[derive(Clone, Debug, Default, Deserialize, JsonSchema, Serialize)]
 pub struct KafkaClusterStatus {}
 
-#[derive(Clone, Debug, Deserialize, JsonSchema, PartialEq, Serialize)]
+#[derive(Clone, Debug, Deserialize, Eq, JsonSchema, PartialEq, Serialize)]
 #[serde(rename_all = "camelCase")]
-pub struct NodeGroup<T> {
-    pub selectors: HashMap<String, SelectorAndConfig<T>>,
-    pub config: Option<T>,
+/// In order for compute_files from the Configuration trait to work, we cannot pass an empty or
+/// "None" config. Therefore we need at least one required property.
+// TODO: Does "log.dirs" make sense in that case? If we make it an option in can happen that
```

Comment:
Sentence incomplete?

Log dirs could potentially be used for this if we have to have a required option. 
I'd usually recommend zookeeper.connect, but that is on its way out, so maybe not the best option..

That being said it seems a weird limitation that we cannot pass an empty config?

Reply:
Yes thats basically a bad interaction between serde (will remove options) and our code. I stumbled over that when implementing this, not in the operator-rs. We definitely should fix that.